app/controllers/admin/courses_controller.rb

```diff
@@ -6,4 +6,17 @@ def index
   def show
     @course =  Course.find(params[:id])
   end
+
+  def add
+    @subscription = Subscription.find(params[:id])
+    @course = Course.find(params[:course_id])
+    return redirect_to [:admin, @subscription], alert: 'Curso não encontrado' unless @course
+
+    if @subscription.courses.include?(@course)
+      redirect_to [:admin, @subscription], alert: 'Curso já cadastrado nesta assinatura'
+    else
+      @subscription.courses << @course
+      redirect_to [:admin, @subscription], alert: 'Curso cadastrado com sucesso'
+    end
```

Comment:
Eu acho que essa action tem lógica demais, como na view é um select você não precisa tratar tudo que chegar. Porque se a pessoa executar algo malicioso ela é a responsável.

Por exemplo:

```ruby
 @course = Course.find(params[:course_id])
return redirect_to [:admin, @subscription], alert: 'Curso não encontrado' unless @course
```

Essa segunda linha nunca vai acontecer porque se o find falhar sobe um erro e não existe um rescue então já daria erro 500 na tela. E não precisa tratar isso porque se a pessoa usar o select bonitinho ela não vai receber esse erro.

```ruby
 if @subscription.courses.include?(@course)
  redirect_to [:admin, @subscription], alert: 'Curso já cadastrado nesta assinatura'
```

Essa próxima você pode tratar com duas verificação uma é quando for preencher os cursos no select não mandar cursos que já estão nessa assinatura, algo como `@courses = Course.where.not(subscription: @subscription)`

A segunda seria adicionar no banco um index único entre essas duas tabelas:

`add_index :subscription_courses, [:couse_id, :subscription_id], unique: true`

Dessa forma um curso nunca pode estar duas vezes na mesma subscription (lembre-se de fazer testes de model)

Isso simplificaria essa action pra basicamente duas linhas e deixa seu controller com muito menos acoplamento